erc1155/src/lib.rs

```diff
@@ -0,0 +1,459 @@
+const ZERO_ID: ActorId = ActorId::new([0u8; 32]);
+
+#[derive(Debug)]
+struct Erc1155Token {
+    name: String,
+    symbol: String,
+    description: String,
+    base_uri: String,
+    balances: BTreeMap<u128, BTreeMap<ActorId, u128>>,
+    operator_approvals: BTreeMap<ActorId, BTreeMap<ActorId, bool>>,
+}
+
+static mut ERC1155_TOKEN: Erc1155Token = Erc1155Token {
+    name: String::new(),
+    symbol: String::new(),
+    base_uri: String::new(),
+    description: String::new(),
+    balances: BTreeMap::new(),
+    operator_approvals: BTreeMap::new(),
+};
+
+impl Erc1155Token {
+    fn balance_of(&self, account: &ActorId, id: &u128) -> u128 {
+        *self
+            .balances
+            .get(id)
+            .and_then(|m| m.get(account))
+            .unwrap_or(&0)
+    }
+
+    fn _set_balance(&mut self, account: &ActorId, id: &u128, amount: u128) {
+        debug!(
+            "before mint: {:?}, id: {:?}",
+            self.balance_of(account, id),
+            id
+        );
+
+        let mut _balance = self
+            .balances
+            .entry(*id)
+            .or_default()
+            .insert(*account, amount);
+
+        debug!(
+            "after mint: {:?}, id: {:?}",
+            self.balance_of(account, id),
+            id
+        );
+    }
+
+    fn balance_of_batch(&self, accounts: &[ActorId], ids: &[u128]) -> Vec<BalanceOfBatchReply> {
+        if accounts.len() != ids.len() {
+            panic!("ERC1155: accounts and ids length mismatch")
+        }
+
+        let mut arr: Vec<BalanceOfBatchReply> = Vec::new();
+
+        for (i, ele) in ids.iter().enumerate() {
+            let account = accounts[i];
+            let amount = self.balance_of(&account, &ele);
+
+            let obj = BalanceOfBatchReply {
+                account: account,
+                id: *ele,
+                amount: amount,
+            };
+
+            arr.push(obj);
+        }
+
+        return arr;
+    }
+
+    fn mint(&mut self, account: &ActorId, id: &u128, amount: u128) {
+        if account == &ZERO_ID {
+            panic!("ERC1155: Mint to zero address")
+        }
+        let old_balance = self.balance_of(account, id);
+        self._set_balance(account, id, old_balance.saturating_add(amount));
+
+        // TransferSingle event
+    }
+
+    fn mint_batch(&mut self, account: &ActorId, ids: &[u128], amounts: &[u128]) {
+        if account == &ZERO_ID {
+            panic!("ERC1155: Mint to zero address")
+        }
+
+        if ids.len() != amounts.len() {
+            panic!("ERC1155: ids and amounts length mismatch")
+        }
+
+        for (i, ele) in ids.iter().enumerate() {
+            let amount = amounts[i];
+            let old_balance = self.balance_of(account, ele);
+            self._set_balance(account, ele, old_balance.saturating_add(amount));
+        }
+
+        // TransferBatch event
+    }
+
+    fn burn_batch(&mut self, ids: &[u128], amounts: &[u128]) {
+        let owner = &msg::source();
```

Comment:
Get the owner of `ids` from the mapping  and check that that owner is `msg_source()`

Reply:
fixed https://github.com/gear-tech/apps/commit/af0f12634f723d208f7de088ef4bacb910a139e0